linkerd/core/src/main/scala/io/buoyant/linkerd/DefaultInterpreterInitializer.scala

```diff
@@ -48,7 +50,14 @@ case class ConfiguredNamersInterpreter(namers: Seq[(Path, Namer)])
       // For now, punt to the default interpreter. Later, we
       // should consider removing the finagle Namer.global, which
       // provides /$/ names.
-      case Nil => Namer.global.lookup(path)
+      case Nil =>
+        Namer.global.lookup(path).transform {
+          case Activity.Failed(e) if e.isInstanceOf[ClassNotFoundException] =>
+            Activity.exception(UnknownNamer(path, e))
+          case Activity.Failed(e) => Activity.exception(e)
```

Comment:
I'm not an activity expert, but transform expects matches on `Activity.State` that map to `Activity` :|